modules/ServerAPI/src/client/SaltBrowser.coffee

```diff
@@ -280,113 +382,211 @@ class SaltBrowserController extends Backbone.View
+		@$(".bv_saltDeletedSuccessfullyMessage").hide()
+		$(".bv_confirmDeleteSalt").show()
 		$('.bv_confirmDeleteSalt').modal({
 			keyboard: false,
 			backdrop: true
 		})
 
-		# NEED TO CHECK DEPENDENCIES 
 		# Clicking Delete should trigger a dependency check. 
-			# Call Dependency Check
-				# See Route Brian Bolt Gave 
 		# If any Lots still reference this salt, then the Salt cannot be deleted.
-			# If Dependency Then Show Preconfigured "Cannot Delete Salt Message"
-			# Show Error Window! 
 		# If there are no dependent lots, then allow the user to confirm and proceed with the delete
-			# If No Dependency Then Proceed w/ Normal Base Implementation 
-			# Show Normal Window 
 
-	# Not Implemented Fully Yet
 	handleConfirmDeleteSaltClicked: =>
-		@$(".bv_deleteWarningMessage").addClass "hide"
-		@$(".bv_deletingStatusIndicator").removeClass "hide"
-		@$(".bv_deleteButtons").addClass "hide"
+		@$(".bv_deleteWarningMessage").hide()
+		@$(".bv_deletingStatusIndicator").show()
+		@$(".bv_deleteButtons").hide()
 
 		$.ajax(
-			url: "/api/cmpdRegAdmin/salts/#{@saltController.model.get("id")}", # NEED TO CHECK VALID ROUTE
+			url: "/api/cmpdRegAdmin/salts/#{@saltController.model.get("id")}", 
 			type: 'DELETE', 
 			success: (result) =>
-				#@$(".bv_okayButton").removeClass "hide"
-				@$(".bv_deletingStatusIndicator").addClass "hide"
-				@$(".bv_saltDeletedSuccessfullyMessage").removeClass "hide"
-				@doSearch "*"
-			error: (result) =>
-				#@$(".bv_okayButton").removeClass "hide"
-				@$(".bv_deletingStatusIndicator").addClass "hide"
-				@$(".bv_errorDeletingSaltMessage").removeClass "hide"
+				@$(".bv_deleteSaltStatus").show()
+				@$(".bv_deletingStatusIndicator").hide()
+				@$(".bv_saltDeletedSuccessfullyMessage").show()
+				@searchController.doSearch("*")
+			error: (errorMsg) =>
+				@$(".bv_deleteSaltStatus").show()
+				@$(".bv_deletingStatusIndicator").hide()
+				@$(".bv_errorDeletingSaltMessage").show()
+				@$(".bv_deleteErrorMessage").html errorMsg
 		)
 
 	handleCancelDeleteClicked: =>
 		@$(".bv_confirmDeleteSalt").modal('hide')
 
+	handleOkayDeleteButtonClicked: =>
+		@$(".bv_confirmDeleteSalt").hide()
+		@$(".bv_deleteSaltStatus").hide()
+
 	handleEditSaltClicked: =>
-		# Need to Implement
-		console.log("Hit Edit Button!")
 		@$('.bv_editSaltWindow').show()
 
+		currAbbrev = @saltController.model.get("abbrev")
+		@$('.bv_abbrevNameEdit').val currAbbrev
+
+		currName = @saltController.model.get("name")
+		@$('.bv_saltNameEdit').val currName 
+
 		molStr = @saltController.model.get("molStructure")
-		console.log(molStr)
 
-		# @$('.bv_editChemicalStructureForm').attr('src',"/lib/ketcher-2.0.0-alpha.3_custom/ketcher.html?api_path=/api/cmpdReg/ketcher/")
-		# @$('.bv_editChemicalStructureForm').on('load', =>
-		# 	@ketcher = @$('.bv_editChemicalStructureForm')[0].contentWindow.ketcher
-		# 	@ketcher.setMolecule(molStr)
-		# )
 		@chemicalStructureController = new KetcherChemicalStructureController 
```

Comment:
I think this was already covered elsewhere, but noting that we'll want to substitute this out for a generic sketcher controller so we can support other sketchers besides Ketcher.